tests/shared/src/test/scala/org/http4s/SchemeSuite.scala

```diff
@@ -27,27 +27,9 @@ import org.http4s.util.Renderer
 import org.scalacheck.Prop._
 
 class SchemeSuite extends munit.DisciplineSuite {
-  test("equals should be consistent with equalsIgnoreCase of the values") {
-    forAll { (a: Scheme, b: Scheme) =>
-      (a == b) == a.value.equalsIgnoreCase(b.value)
-    }
-  }
-
-  test("compare should be consistent with value.compareToIgnoreCase") {
-    forAll { (a: Scheme, b: Scheme) =>
-      assertEquals(a.value.compareToIgnoreCase(b.value), a.compare(b))
-    }
-  }
-
-  test("hashcode should be consistent with equality") {
-    forAll { (a: Scheme, b: Scheme) =>
-      (a == b) ==> (a.## == b.##)
-    }
-  }
```

Comment:
These tests are just asserting that a `CIString` behaves correctly. I agree that these are valuable tests, that belong in the `case-insensitive` project :)

Reply:
These tests guarantee the behaviour the code must have. Implementation could be changed over time (as it's done in this PR particularly).

Reply:
If they at least were affecting the execution time of the tests... but they are almost free, are they?

Reply:
I will reintroduce the tests. The reason for removing them was, that I found no equivalent tests for other very similar case where CIString is wrapped. But I agree that they re very fast to run and are not unhelpful.

Reply:
> > These tests guarantee the behaviour the code must have.
> 
> Good point. I think it could make sense to add checks for `Hash` laws from Cats, this will test that equality is consistent with hash etc.

I think that would require a `Hash`  instance for `Scheme` which I think is currently not implemented. Adding it to allow for HashLaw testing would be possible of course. I think I prefer just undeleteting and fix the old tests for now. wdyt?

Reply:
undeleted and fixed. If you think the HashLaws tests are they way to go, please leave comment and I can have another look :)